specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/managedHsm.json

```diff
@@ -0,0 +1,634 @@
+        },
+        "securityDomainId": {
+          "type": "string",
+          "format": "uuid",
+          "description": "The security domain id that should be used for restoring data from this managed HSM pool to a new managed hsm pool."
+        },
+        "initialAdminObjectIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Array of initial administrators object ids for this managed hsm pool."
+        },
+        "hsmPoolUri": {
+          "type": "string",
+          "description": "The URI of the managed hsm pool for performing operations on keys and secrets."
+        },
+        "enableSoftDelete": {
+          "type": "boolean",
+          "default": true,
+          "description": "Property to specify whether the 'soft delete' functionality is enabled for this managed HSM pool. If it's not set to any value(true or false) when creating new key vault, it will be set to true by default. Once set to true, it cannot be reverted to false."
+        },
+        "softDeleteRetentionInDays": {
+          "type": "integer",
+          "format": "int32",
+          "default": 90,
+          "description": "softDelete data retention days. It accepts >=7 and <=90."
+        },
+        "enablePurgeProtection": {
+          "type": "boolean",
+          "description": "Property specifying whether protection against purge is enabled for this managed HSM pool. Setting this property to true activates protection against purge for this managed HSM pool and its content - only the Managed HSM service may initiate a hard, irrecoverable deletion. The setting is effective only if soft delete is also enabled. Enabling this functionality is irreversible - that is, the property does not accept false as its value."
+        },
+        "createMode": {
+          "type": "string",
+          "description": "The create mode to indicate whether the resource is being created or is being recovered from a deleted resource.",
+          "enum": [
+            "recover",
+            "default"
+          ],
+          "x-ms-enum": {
+            "name": "CreateMode",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "recover",
+                "description": "Recover the vault from a soft-deleted resource."
+              },
+              {
+                "value": "default",
+                "description": "Create a new vault.  This is the default option."
+              }
+            ]
+          }
+        },
+        "statusMessage": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Status Message."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state.",
+          "enum": [
```

Comment:
Based on Amit's spec shouldn't these be:

- Provisioning
- Provisioned
- Activated
- SecurityDomainRestore
- Restoring
- SoftDeleted